ports/nrf/mphalport.c

```diff
@@ -35,6 +35,113 @@
 #include "nrfx_errors.h"
 #include "nrfx_config.h"
 
+#if MICROPY_PY_TIME_TICKS
+#include "nrfx_rtc.h"
+#include "nrf_clock.h"
+#endif
+
+
+#if MICROPY_PY_TIME_TICKS
+// use RTC1 for time ticks generation
+// with 32kHz tick resolution and overflow
+// handling in irq
+
+nrfx_rtc_t rtc1 = NRFX_RTC_INSTANCE(1);
+volatile mp_uint_t rtc_overflows = 0;
+
+const nrfx_rtc_config_t rtc_config_time_ticks = {
+    .prescaler = 0,
+    .reliable = 0,
+    .tick_latency = 0,
+    #ifdef NRF51
+    .interrupt_priority = 1,
+    #else
+    .interrupt_priority = 3,
+    #endif
+};
+
+STATIC void rtc_irq_time(nrfx_rtc_int_type_t event) {
+    // irq handler for overflow
+    if (event == NRFX_RTC_INT_OVERFLOW) {
+        rtc_overflows += 1;
+    }
+    // irq handler for wakeup from WFI (~1msec)
+    if (event == NRFX_RTC_INT_COMPARE0) {
+        nrfx_rtc_cc_set(&rtc1, 0, nrfx_rtc_counter_get(&rtc1) + 33, true);
```

Comment:
the "33" should probably be a #define constant

Reply:
yes